### ISSUE TITLE
"inappropriate type (not a ghc-pkg db file)"

### ISSUE THREAD

**Issue by thomasjm** [commercialhaskell/stack-ide#42]
I run `stack ide start` in my project and I get the following error:

```
{"tag":"ResponseWelcome","contents":[0,1,1]}
{"tag":"ResponseUpdateSession","contents":{"tag":"UpdateStatusCrashRestart","contents":"External exception: /Users/tomm/.stack/snapshots/x86_64-osx/lts-2.16/7.8.4/pkgdb/package.cache: GHC.PackageDb.readPackageDb: inappropriate type (not a ghc-pkg db file, wrong file magic number). Local exception: user error (IdeSession.RPC.Stream ended, causing demandInput: not enough bytes)"}}
```

**Comment by mgsloan** [comment]
My guess is that the `ide-backend-server` on your path is using a newer version of GHC.  `ResponseWelcome` really ought to give more info about GHC versions and such.  I've created this issue to track that: #43

`stack ide` does not yet streamline making sure that the right version of ide-backend-server is used, and does not yet automate its installation.  So for now, that part of things is manual.  Relevant stack issue: https://github.com/commercialhaskell/stack/issues/662

If I'm right about the cause, there are two approaches to resolve this:
- Run `stack --stack-yaml=stack-7.8.yaml install` in the ide-backend or stack-ide repo.  This will install a version that uses ghc 7.8.
- Run `stack install` in the ide-backend or stack-ide repo.  Update your project to use a very recent nightly resolver (such as `nightly-2015-08-06`), so that you're using ghc-7.10.2 for your project.

(NOTE: you'll also want to pull both of these repos, I updated all the stack.yaml resolvers a few minutes ago)
